source/extensions/access_loggers/filters/local_ratelimit/filter.cc

```diff
@@ -0,0 +1,41 @@
+#include "source/extensions/access_loggers/filters/local_ratelimit/filter.h"
+
+#include "envoy/access_log/access_log.h"
+
+#include "source/common/init/target_impl.h"
+#include "source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace AccessLoggers {
+namespace Filters {
+namespace LocalRateLimit {
+LocalRateLimitFilter::LocalRateLimitFilter(
+    Server::Configuration::FactoryContext& context,
+    const envoy::extensions::access_loggers::filters::local_ratelimit::v3::LocalRateLimitFilter&
+        config) {
+  rate_limiter_ = Envoy::Extensions::Filters::Common::LocalRateLimit::RateLimiterProviderSingleton::
+      getRateLimiter(
+          context, config.resource_name(), config.config_source(),
+          [this](std::shared_ptr<
+                 Envoy::Extensions::Filters::Common::LocalRateLimit::LocalRateLimiterImpl>
+                     limiter) -> void {
+            ASSERT(limiter != nullptr, "limiter shouldn't be null if the `limiter` is set from "
+                                       "callback.");
+            rate_limiter_->setLimiter(limiter);
```

Comment:
This is rather confusing setup - it seems to modify itself. Maybe move the limiter itself as a separate top-level field?

Reply:
Renamed. I still wanna have a wrapper class to hold the <RateLimiter, ProviderSingleton> together.